src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -2078,7 +2078,7 @@ internal static BinaryOperationKind DeriveBinaryOperationKind(UnaryOperationKind
                     return BinaryOperationKind.DynamicSubtract;
 
                 default:
-                    throw ExceptionUtilities.UnexpectedValue(incrementKind);
+                    return BinaryOperationKind.None;
```

Comment:
If we were to add Invalid, its value would need to be 0, for reasons that will become clear when you see my upcoming API review PR. Maybe leave it as None for today.